rusk-wallet/src/lib/wallet.rs

```diff
@@ -121,64 +152,83 @@ impl CliWallet {
                 gas_limit,
                 gas_price,
             } => {
-                let my_addr = self.wallet.public_spend_key(key)?;
-                let mut rng = StdRng::from_entropy();
-                self.wallet.stake(
-                    &mut rng,
-                    key,
-                    stake_key,
-                    &my_addr,
-                    amt,
-                    gas_limit,
-                    gas_price.unwrap_or(0),
-                )?;
-                println!("Stake success! ✅");
-            }
-
-            // Extend stake for a particular key
-            ExtendStake {
-                key,
-                stake_key,
-                gas_limit,
-                gas_price,
-            } => {
-                let my_addr = self.wallet.public_spend_key(key)?;
-                let mut rng = StdRng::from_entropy();
-                self.wallet.extend_stake(
-                    &mut rng,
-                    key,
-                    stake_key,
-                    &my_addr,
-                    gas_limit,
-                    gas_price.unwrap_or(0),
-                )?;
-                println!("Stake extension success! ✅");
+                if let Some(wallet) = &self.wallet {
+                    let my_addr = wallet.public_spend_key(key)?;
+                    let mut rng = StdRng::from_entropy();
+                    wallet.stake(
+                        &mut rng,
+                        key,
+                        stake_key,
+                        &my_addr,
+                        amt,
+                        gas_limit,
+                        gas_price.unwrap_or(0),
+                    )?;
+                    println!("> Stake success!");
+                    Ok(())
+                } else {
+                    Err(Error::Offline)
+                }
             }
-
+            /*
+                        // Extend stake for a particular key
+                        ExtendStake {
+                            key,
+                            stake_key,
```

Comment:
I'm not a fan of commented code... if we need it back we have the git logs

Reply:
I have no problem in yanking it completely, but I asked @autholykos explicitly about this and we agreed that for now we would just hide this from the user.